src/EchoToImageLUT.cpp

```diff
@@ -134,8 +130,7 @@ void EchoToImageLUT::drawImageFromEchoes(std::vector<uint8_t> const& world_echoe
 pair<vector<Point>::const_iterator, vector<Point>::const_iterator> EchoToImageLUT::
     getPixels(int angle_idx, int sweep_idx) const
 {
-    int initial_point_idx = m_data_index[angle_idx * m_sweep_size + sweep_idx];
-    int final_point_idx = m_data_index[angle_idx * m_sweep_size + sweep_idx + 1] - 1;
-    return std::make_pair(m_data.cbegin() + initial_point_idx,
-        m_data.cbegin() + final_point_idx);
+    int pixels_begin = m_data_index[angle_idx * m_sweep_size + sweep_idx];
+    int pixels_end = m_data_index[angle_idx * m_sweep_size + sweep_idx + 1];
```

Comment:
It is not actually sweep index, is the echo dot index
```suggestion
    int pixels_begin = m_data_index[angle_idx * m_sweep_size + echo_dot_idx];
    int pixels_end = m_data_index[angle_idx * m_sweep_size + echo_dot_idx + 1];
```

Reply:
IMO it is indeed is the sweep idx. The range is [0, sweep_length[